CONTRIBUTING.md

```diff
@@ -1,94 +1,121 @@
+- **DO** read our [voice and tone](voice-tone.md) guidelines.
+- **DO** use our [template](template.md) file as the starting point of your work. This file also contains our style guide guidelines.
+- **DO** create a separate branch on your fork before working on the articles.
+- **DO** follow the [GitHub Flow workflow](https://guides.github.com/introduction/flow/). 
+- **DO** blog and tweet (or whatever) about your contributions, frequently!
 
-	docfx build
-	
-After build completes, you can preview the resulting site locally using built-in web server.
+## Building the docs ##
+
+The documentation is built using [DocFX](http://dotnet.github.io/docfx/), 
+[Markdown](https://daringfireball.net/projects/markdown/syntax) and other internal publishing/building tools. It is hosted at [docs.microsoft.com/dotnet](https://docs.microsoft.com/dotnet). 
 
-	docfx serve _site
+To build the docs locally, you need to install [DocFX](https://dotnet.github.io/docfx/); latest versions are the best.
+
+There are several ways to use DocFX, and most of them are covered in the [DocFX getting started guide](https://dotnet.github.io/docfx/tutorial/docfx_getting_started.html). 
+The following instructions use the [command-line based](https://dotnet.github.io/docfx/tutorial/docfx_getting_started.html#2-use-docfx-as-a-command-line-tool) version of the tool. 
+If you are comfortable with other ways listed on the link above, feel free to use those. 
+
+**Note:** please note that as of now, DocFX requires the .NET Framework on Windows or Mono for Linux or macOS. We'll be working towards porting it to .NET Core in the future. 
+
+You can build and preview the resulting site locally using a built-in web server. Navigate to the core-docs folder on your machine and type the following command:
+
+	docfx .\docfx.json --serve
 	
-This will start the local preview on localhost:8080. You can then view the changes by going to http://localhost:8080/docs/[path] (e.g. http://localhost:8080/docs/getting-started/).   
+This starts the local preview on [localhost:8080](http://localhost:8080). You can then view the changes by going to `http://localhost:8080/docs/[path]`, such as http://localhost:8080/docs/articles/.
+
+**Note:** the local preview currently doesn't contain any themes at the moment so the look and feel won't be the same as in our site. We're working towards fixing that experience.
 
 ## Contributing to samples
 
 See the [Samples Readme](https://github.com/dotnet/core-docs/blob/master/samples/README.md).
+
+##Contributor License Agreement
+
+You must sign a [.NET Foundation Contribution License Agreement (CLA)](http://cla2.dotnetfoundation.org) before your PR will be merged. This a one-time requirement for projects in the .NET Foundation. You can read more about [Contribution License Agreements (CLA)](http://en.wikipedia.org/wiki/Contributor_License_Agreement) on Wikipedia.
```

Comment:
Good catch. I just copied and pasted from the main repo, so I'll open a PR for this change there too.